pycbc/io/hdf.py

```diff
@@ -803,27 +814,21 @@ def get_active_segments(self):
                                                                    ends)
 
     def get_end_time(self):
-        ifos = self.coinc_file.h5file.attrs['ifos'].split(' ')
         times_gen = (self.get_coincfile_array('{}/time'.format(ifo))
-                     for ifo in ifos)
+                     for ifo in self.ifos)
         ref_times = np.array([mean_if_greater_than_zero(t)[0]
                               for t in zip(*times_gen)])
         return ref_times
 
     def get_ifos(self):
+        # FIXME: Explain what this does
```

Comment:
This seems to identify which ifos are participating in each coincidence. So if you have 4 coincidences, one that's HLV, one that's HL, one HV and one just H it should return:

`[['H1', 'L1', 'V1'], ['H1', 'L1'], ['H1', 'V1'], ['H1']]`

(At least that's my reading of this). It uses the condition that we store a negative time to indicate that an ifo did not participate in coincidence, but given that this is using a [slow] for loop on line 830, I don't really see why it doesn't do this in one direct for loop. Something like:

* Get the arrays of times for all ifos
* Loop over all triggers, identify which times are negative for each trigger, construct and append the correct ifo string.

.... But changing that doesn't seem in scope of this PR.

Reply:
I tried doing print(fortrigs.get_ifos()) and got something which looked like a list of lists of coincident ifos, but all of them had exactly 2 ifos which seems wrong. Git blame tells me this was Fergus' code and git grep tells me that the get_ifos method is not used for anything (there is a method with the same name in pygrb postproc utils but that seems to be inherited from lsctables!)

Reply:
So given that we don't know if this code is correct, a FIXME may be warranted